test/vfd.c

```diff
@@ -140,6 +140,13 @@ static int __k;
         }                                                                                                    \
     } while (FALSE)
 
+/*
+ * Ugly hack to cast away const for freeing const-qualified memory
+ */
+#ifndef HDfree_const
+#define HDfree_const(mem) HDfree((void *)(uintptr_t)mem)
+#endif
```

Comment:
We should put this in H5private.h.

Reply:
I do hate this hack, but I thought about putting it in H5private.h. I don't want to encourage people to do this, but it is kind of necessary in some cases. I can put it in H5private and leave a stern comment.

Reply:
I see your original comment now Dana, I feel like I'm more on Jordan's side on that, it was better to leave it in that particular test imo. I don't think encouraging that kind of hack is a good thing :)